lightning-persister/src/utils.rs

```diff
@@ -0,0 +1,126 @@
+use std::fs;
+use std::path::{Path, PathBuf};
+
+#[cfg(not(target_os = "windows"))]
+use std::os::unix::io::AsRawFd;
+
+pub(crate) trait DiskWriteable {
+	fn write_to_file(&self, writer: &mut fs::File) -> Result<(), std::io::Error>;
+}
+
+pub(crate) fn get_full_filepath(filepath: String, filename: String) -> String {
+	let mut path = PathBuf::from(filepath);
+	path.push(filename);
+	path.to_str().unwrap().to_string()
+}
+
+pub(crate) fn write_to_file(path: String, filename: String, data: &dyn DiskWriteable) -> std::io::Result<()> {
+	fs::create_dir_all(path.clone())?;
+	// Do a crazy dance with lots of fsync()s to be overly cautious here...
+	// We never want to end up in a state where we've lost the old data, or end up using the
+	// old data on power loss after we've returned.
+	// The way to atomically write a file on Unix platforms is:
+	// open(tmpname), write(tmpfile), fsync(tmpfile), close(tmpfile), rename(), fsync(dir)
+	let filename = get_full_filepath(path, filename);
```

Comment:
Its really confusing that you alias the filename argument with a filename variable here.